lib/appsignal/rack/generic_instrumentation.rb

```diff
@@ -16,7 +16,8 @@ def call(env)
         if Appsignal.active?
           call_with_appsignal_monitoring(env)
         else
-          @app.call(env)
+          status, headers, obody = @app.call(env)
+          [status, headers, Appsignal::Rack::BodyWrapper.wrap(obody, _transaction = nil)]
```

Comment:
A pattern we use is using [`Appsignal::Transaction::NilTransaction`](https://rubydoc.info/gems/appsignal/3.5.6/Appsignal/Transaction/NilTransaction) when no transaction is active when calling `Appsignal::Transaction.current`. Appsignal::Transaction::NilTransaction is * not actually a nil object *, but responds to any method call and ignores it. Which is great for when you don't want your implementation in BodyWrapper to constantly check if the transaction variable is not nil.

Would that work here too, and in the other middlewares where the same thing happens? Pass in `Appsignal::Transaction::NilTransaction.new` and remove all the nil-checks from the BodyWrapper implementation.

Reply:
We could use the `NilTransaction`, sure. What threw me off there was that there would be checks anyway because I don't know what will happen when you call `complete_current!` (or rather: it is possible to find out what happens, but I don't know whether that behaviour is stable) - so `transaction.complete` cannot be used, which would have been nice. If we can remove checks entirely and `complete_current!` will do the thing gracefully - even better.